addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -820,7 +820,11 @@ function makeActionManager(env) {
      * @param {ActionOptions} options
      */
     async function _executeClientAction(action, options) {
-        const clientAction = actionRegistry.get(action.tag);
+        let clientAction = actionRegistry.get(action.tag);
+        if (typeof clientAction === "function") {
+            // this is a lazy action, we can replace it with its executed version.
+            clientAction = clientAction();
+        }
```

Comment:
This is already handled by the the else line 843: if the client action is not a Component it is expected to be a function, which is called and awaited, the resulting action is fed to doAction. This should cover your case and more (eg the concrete action can be determined from an RPC to the server, and doesn't have to be a client action).